core/src/main/java/io/grpc/internal/AbstractManagedChannelImplBuilder.java

```diff
@@ -197,7 +211,11 @@ public final T nameResolverFactory(NameResolver.Factory resolverFactory) {
     Preconditions.checkState(directServerAddress == null,
         "directServerAddress is set (%s), which forbids the use of NameResolverFactory",
         directServerAddress);
-    this.nameResolverFactory = resolverFactory;
+    if (resolverFactory != null) {
```

Comment:
I'm purposefully avoiding changing the externally-visible behavior of this class. The previous code allowed passing in `null` as a way to choose the default, so the new code maintains that behavior.